### PR TITLE
Convert shield.js to typescript

### PR DESCRIPTION
**Pull request by ZeLonewolf**
This PR converts the remaining code file in the shield library implementation to typescript.  This refactor caught several bugs and allowed me to remove some dead that I discovered in the refactor.  This gives me a bit more confidence that the stronger typing in typescript is worthwhile/desirable.

### CODE REVIEW

File: shieldlib/src/shield.ts [hunk 1]
```diff
@@ -108,23 +113,23 @@ function drawShieldText(r, ctx, shieldDef, routeDef, shieldBounds) {
   ctx.fillStyle = textColor(shieldDef);
   ShieldText.renderShieldText(r, ctx, routeDef.ref, textLayout);
 
-  if (r.options.SHIELD_TEXT_BBOX_COLOR) {
-    ctx.strokeStyle = r.options.SHIELD_TEXT_BBOX_COLOR; //TODO move to debugOptions
+  if (r.debugOptions?.shieldTextBboxColor) {
+    ctx.strokeStyle = r.debugOptions.shieldTextBboxColor; //TODO move to debugOptions
```
Comment (claysmalley) [inline shieldlib/src/shield.ts:117]: Now that this is moved to debugOptions, can we get the debug options to work again?

Comment (ZeLonewolf) [reply shieldlib/src/shield.ts:117]: image

A lot easier now in TS where I can figure out what's wrong :smile:

Comment (claysmalley) [reply shieldlib/src/shield.ts]: Fantastic! It's working for me on the main map view, but not on shieldtest.html - any idea why?

Comment (ZeLonewolf) [reply shieldlib/src/shield.ts]: Fixed in bd0993f!

**Review by claysmalley — APPROVED** [review]

**Pull request merged**